### PR TITLE
[go] Add filesystem write fallback for `startDevServer()`

### PR DESCRIPTION
**Pull request by TooTallNate**
This is for `@vercel/go` to work on Windows, which currently fails with
this error:

```
panic: write pipe: The handle is invalid.

goroutine 1 [running]:
main.main()
	C:/Users/Nathan/Code/casca/vercel-go-test/.vercel/cache/go/2flefmhra8o/api/vercel-dev-server-main.go:25 +0x1ec
exit status 2
```

So this fallback writes the port number to a temp file that
`startDevServer()` polls for in order to figure out the port number.

### CODE REVIEW

File: packages/now-go/index.ts [hunk 1]
```diff
@@ -562,19 +567,56 @@ Learn more: https://vercel.com/docs/runtimes#official-runtimes/go`
...
       pid: child.pid,
     };
-  } else {
+  } else if (Array.isArray(result)) {
     // Got "exit" event from child process
     throw new Error(
       `Failed to start dev server for "${entrypointWithExt}" (code=${result[0]}, signal=${result[1]})`
     );
+  } else {
+    throw new Error('Unexpected error');
```
Comment (styfle) [inline packages/now-go/index.ts]:
```suggestion
    throw new Error(`Unexpected result type: ${typeof result}`);
```

File: packages/now-go/index.ts [hunk 2]
```diff
@@ -562,19 +567,56 @@ Learn more: https://vercel.com/docs/runtimes#official-runtimes/go`
...
+  }
+}
+
+export interface CancelablePromise<T> extends Promise<T> {
+  cancel: () => void;
+}
```
Comment (styfle) [inline packages/now-go/index.ts:593]: I wish Node.js had [AbortController](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/AbortController) 😖

**Review by styfle — DISMISSED** [review]

File: packages/now-go/index.ts (hunk 2 above)
Comment (TooTallNate) [reply packages/now-go/index.ts:593]: Deno 😉

File: packages/now-go/dev-server.go [hunk 3]
```diff
@@ -12,17 +13,23 @@ func main() {
 	handler := http.HandlerFunc(__HANDLER_FUNC_NAME)
 
 	// https://stackoverflow.com/a/43425461/376773
-	listener, err := net.Listen("tcp", ":0")
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
```
Comment (TooTallNate) [inline packages/now-go/dev-server.go:16]: Binding only to localhost prevents a firewall prompt on Windows

**Review by AndyBitz — APPROVED** [review]

**Review by styfle — APPROVED** [review]

**Comment by cascandaliato** [comment]
Thank you for fixing this issue so quickly.

@paulogdm gave me the link to this PR and I am amazed by how transparent and streamlined the entire process is, from support ticket to code reviews, comments, and CI/CD.

I fell in love with the workflow shown in this PR page.

**Pull request merged**